publish-token/src/main/java/fi/thl/covid19/publishtoken/generation/v1/PublishTokenGenerationRequest.java

```diff
@@ -23,12 +24,22 @@ public PublishTokenGenerationRequest(String requestUser,
                                          LocalDate symptomsOnset,
                                          Optional<String> patientSmsNumber,
                                          Optional<Boolean> validateOnly) {
-        this.requestUser = validateUserName(requireNonNull(requestUser, "User required"));
-        this.symptomsOnset = requireNonNull(symptomsOnset, "Symptoms onset date required");
-        this.patientSmsNumber = requireNonNull(patientSmsNumber).map(Validation::normalizeAndValidatePhoneNumber);
-        if (symptomsOnset.isAfter(LocalDate.now(ZoneOffset.UTC).plus(1, ChronoUnit.DAYS))) {
-            throw new InputValidationException("Symptoms onset time in the future: date=" + symptomsOnset);
-        }
+
         this.validateOnly = validateOnly.orElse(false);
+
+        try {
+            this.requestUser = validateUserName(requireNonNull(requestUser, "User required"));
+            this.symptomsOnset = requireNonNull(symptomsOnset, "Symptoms onset date required");
+            this.patientSmsNumber = requireNonNull(patientSmsNumber).map(Validation::normalizeAndValidatePhoneNumber);
+            if (symptomsOnset.isAfter(LocalDate.now(ZoneOffset.UTC).plus(1, ChronoUnit.DAYS))) {
+                throw new InputValidationException("Symptoms onset time in the future: date=" + symptomsOnset);
+            }
+        } catch (InputValidationException | NullPointerException e) {
+            if (this.validateOnly) {
+                throw new InputValidationValidateOnlyException(e.getMessage());
```

Comment:
The original exception could be put in as the cause -throwable, so that the stack (if ever used) is correct and points to the row where the problem happened. In this case, it doesn't matter of course, because we're never going to log that stack.